website/docs/getting-started/basics.md

```diff
@@ -8,15 +8,34 @@ odo abstracts Kubernetes concepts into developer focussed terminology. In this d
 
 ### Concepts
 
-#### Component
+* Component
+  A component is a microservice. Each component can be run and deployed separately. Examples of components: Warehouse API Backend, Inventory API, Web Frontend, Payment Backend
+  > A component in odo is an equivalent of a deployment or deploymentconfig in Kubernetes. Creating a new component in odo will create a deployment in Kubernetes or for git and binary based component creation, it will create deploymentconfig.
 * Application
+  An application consists of multiple components which may span over multiple projects, and work individually to build the entire application. Examples of applications: e-Shop, Hotel Reservation System, Online Booking
+  > An application can be considered as an equivalent of labels in Kubernetes that help in grouping a set of resources.
 * Project
+  A project is your source code, tests, and libraries organized in a separate single unit.
```

Comment:
This sounds more like a definition of what a component directory (a.ka. context) is on the dev's local system.

Reply:
Without naming the underlying Kubernetes `namespace` resource, you can explain the motivations of using separate namespaces:
- a scope for names
- divide cluster resources between users (resources quotas, RBAC are based on namespaces)

https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/

Reply:
Yeah, I actually picked that up from one of the existing docs. I'll try to fix that.